apps/remote_control/lib/lexical/remote_control/code_mod/replace_with_underscore.ex

```diff
@@ -33,55 +25,21 @@ defmodule Lexical.RemoteControl.CodeMod.ReplaceWithUnderscore do
     variable_name
   end
 
-  defp apply_transform(line_text, quoted_ast, unused_variable_name) do
+  defp apply_transform(document, line_number, unused_variable_name) do
     underscored_variable_name = :"_#{unused_variable_name}"
-    leading_indent = leading_indent(line_text)
-
-    Macro.postwalk(quoted_ast, fn
-      {^unused_variable_name, meta, nil} ->
-        {underscored_variable_name, meta, nil}
-
-      other ->
-        other
-    end)
-    |> Macro.to_string()
-    # We're dealing with a single error on a single line.
-    # If the line doesn't compile (like it has a do with no end), ElixirSense
-    # adds additional lines do documents with errors, so take the first line, as it's
-    # the properly transformed source
-    |> fetch_line(0)
-    |> case do
-      {:ok, text} ->
-        {:ok, "#{leading_indent}#{text}"}
-
-      error ->
-        error
-    end
-  end
 
-  @indent_regex ~r/^\s+/
-  defp leading_indent(line_text) do
-    case Regex.scan(@indent_regex, line_text) do
-      [indent] -> indent
-      _ -> ""
-    end
-  end
+    result =
+      Ast.traverse_line(document, line_number, [], fn
+        {{^unused_variable_name, _meta, nil} = node, _} = zipper, patches ->
+          [patch] = Sourceror.Patch.rename_identifier(node, underscored_variable_name)
```

Comment:
It looks great here, much simpler.

Reply:
Yes, this was a big improvement in both code readability and ease of use.